todo-list-backend/tests/integration/graphql-api.spec.ts

```diff
@@ -0,0 +1,526 @@
+    ];
+
+    return new InMemoryStorage(initialUsers, initialTodos);
+
+}
+
+describe("query", () => {
+    const authSecret: string = "This is a perfect secret for testing!";
+    let user: User;
+    let storage: IStorage;
+    let query: any;
+    let mutate: any;
+
+
+    describe("given a legit authorization token", () => {
+        beforeEach(async () => {
+            storage = buildTwoUserInMemoryStorage(true);
+
+            // workaround for the shortcomings of the apollo-server-testing
+            // package as it doesn't generate a real request and therefor a
+            // "real" context will never be generated
+            user = storage.readUser("ralph");
+            const token = sign({username: user.name}, authSecret, {expiresIn: "1 day"});
+            const context = {token: token, user: user};
+
+            const testServer = buildApolloServer(storage, authSecret, false, context);
+            const testClient = createTestClient(testServer);
+            query = testClient.query;
+            mutate = testClient.mutate;
+        });
+
+        describe("readTodos", () => {
+            it("returns todos of authorized user", async () => {
+               let response = await query({
+                   query: READ_TODOS
+               });
+
+                expect(response.data.readTodos).toHaveLength(1);
+            });
+        });
+    });
+    describe("given a random authorization token", () => {
+        beforeEach(async () => {
+            storage = buildTwoUserInMemoryStorage(true);
+
+            const context = {token: "I'm not a valid token"};
+
+            const testServer = buildApolloServer(storage, authSecret, false, context);
+
+            const testClient = createTestClient(testServer);
+            query = testClient.query;
+            mutate = testClient.mutate;
+        });
+
+        describe("readTodos", () => {
+            it("raises an error", async () => {
+                let response = await query({
+                    query: READ_TODOS
+                });
+
+                expect(response.errors).toHaveLength(1);
+                expect(response.data).toBeNull();
+            });
+        });
+    });
+});
+
+describe("mutate", () => {
+    const authSecret: string = "This is a perfect secret for testing!";
+
+    describe("given correct user credentials", () => {
+        const storage: IStorage = buildTwoUserInMemoryStorage();
+        const testServer = buildApolloServer(storage, authSecret, false, {});
+        const {query, mutate} = createTestClient(testServer);
+
+        describe("login", () => {
+            it("returns a jwt", async () => {
+                let tokenResponse = await mutate({
+                    mutation: LOGIN,
+                    variables: {
+                        name: "ralph",
+                        password: "ralph"
+                    }
+                });
+
+                expect(tokenResponse.data.login).toBeDefined();
```

Comment:
```suggestion
                expect(tokenResponse.data.login).toEqual(expect.any(String));
```
Avoids `null` and `false` I guess.